UDP/current_term_courses/current_term_courses_matview_bq.sql

```diff
@@ -0,0 +1,208 @@
+DROP MATERIALIZED view v1_student_current_term_course;
+
+create materialized view v1_student_current_term_course AS
+WITH
+ term_info AS (
+ SELECT
+   academic_term_id,
+   name AS term_name
+ FROM
+   `udp-umich-prod.context_store_entity.academic_term` at2
+ WHERE
+   DATE(current_timestamp) < at2.term_end_date
+   AND DATE(current_timestamp) > at2.term_begin_date ),
+ courses AS (
+ SELECT
+   co.course_offering_id,
+   co.le_code AS canvas_course_title,
+   co.available_credits AS credits,
+   co.academic_term_id,
+   term_name
+ FROM
+   `udp-umich-prod.context_store_entity.course_offering` co
+ JOIN
+   term_info at2
+ ON
+   co.academic_term_id = at2.academic_term_id
+ WHERE
+   co.le_status = 'available'),
+ sections AS (
+ SELECT
+   cs.course_section_id,
+   cs2.lms_ext_id AS canvas_section_id,
+   cs.le_current_course_offering_id AS course_offering_id,
+   co4.lms_ext_id AS canvas_course_id,
+   cs.le_name AS section_name,
+   cs.section_number AS section
+ FROM
+   `udp-umich-prod.context_store_entity.course_section` cs
+ JOIN
+   `udp-umich-prod.context_store_keymap.course_section` cs2
+ ON
+   cs.course_section_id = cs2.id
+ JOIN
+   `udp-umich-prod.context_store_keymap.course_offering` co4
+ ON
+   co4.id = cs.le_current_course_offering_id
+   AND cs.status='active'),
+ courses_sections_of_current_term AS (
+ SELECT
+   c.canvas_course_title,
+   c.term_name,
+   s.*,
+   c.academic_term_id,
+   c.credits
+ FROM
+   courses c
+ JOIN
+   sections s
+ ON
+   c.course_offering_id=s.course_offering_id),
+ enrollment AS (
```

Comment:
So this enrollment gets all enrollments from all courses in the database. Can we limit the scope to only current term courses/sections based on above courses_sections_of_current_term table?

Reply:
That's a reasonable request. I will work on it. I think event with this, the query ran very fast like in 1 mins.

Reply:
Done, I removed `enrollment` sub-set and did a join on `courses_sections_of_current_term` and `entity.course_section_enrollment` and then joined on `person` and `person_email`.  I checked with these courses (577751, 582106) which has students in multiple sections to see if it still preserve the student enrolled in multiple section and checking enrollment count.

Reply:
Looks good!